### PR TITLE
add sdist command and umask reminder

### PR DESCRIPTION
**Pull request by esc**
As title

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/first_rc_checklist.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ labels: task
 * [ ] Build and upload conda packages on buildfarm (check "upload").
 * [ ] Build wheels (`$PYTHON_VERSIONS`) on the buildfarm.
 * [ ] Verify packages uploaded to Anaconda Cloud and move to `numba/label/main`.
+* [ ] Build sdist locally using `python setup.py sdist --user=ci --group=numba` with umask `0022`.
```
Comment (stuartarchibald) [inline .github/ISSUE_TEMPLATE/first_rc_checklist.md:25]: Did the umask used get confirmed by @seibert ? xref: https://github.com/numba/numba/issues/7800#issuecomment-1024522300

**Comment by stuartarchibald** [comment]
Note: I think merge of this could reasonably close #7800.

File: .github/ISSUE_TEMPLATE/first_rc_checklist.md (hunk 1 above)
Comment (seibert) [reply .github/ISSUE_TEMPLATE/first_rc_checklist.md:25]: yes, this umask is correct

Comment (stuartarchibald) [inline .github/ISSUE_TEMPLATE/first_rc_checklist.md:25]:
```suggestion
* [ ] Build sdist locally using `python setup.py sdist -u=ci --group=numba` with umask `0022`.
```

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch. Suggest minor change to match output of `python setup.py sdist --help` which gives:
```
  --owner (-u)           Owner name used when creating a tar file [default:
                         current user]
```

File: .github/ISSUE_TEMPLATE/first_rc_checklist.md (hunk 1 above)
Comment (esc) [reply .github/ISSUE_TEMPLATE/first_rc_checklist.md:25]: should we then also use `-g` for coherence?

File: .github/ISSUE_TEMPLATE/sub_rc_checklist.md [hunk 2]
```diff
@@ -19,6 +19,7 @@ labels: task
 * [ ] Verify packages uploaded to Anaconda Cloud and move to
   `numba/label/main`.
 * [ ] Build wheels (`$PYTHON_VERSIONS`) on the buildfarm.
+* [ ] Build sdist locally using `python setup.py sdist --user=ci --group=numba` with umask `0022`.
```
[the remark this answers is not in the document]
Comment (esc) [reply .github/ISSUE_TEMPLATE/sub_rc_checklist.md:22]: should we then also use `-g` for coherence?

File: .github/ISSUE_TEMPLATE/first_rc_checklist.md (hunk 1 above)
Comment (stuartarchibald) [reply .github/ISSUE_TEMPLATE/first_rc_checklist.md:25]: Am happy with either, though think that `-g` does make it more consistent. The most important thing is to fix the `--user` part as it seems like that doesn't exist!

Comment (stuartarchibald) [reply .github/ISSUE_TEMPLATE/first_rc_checklist.md:25]: `--owner` sounds good RE obviousness!

**Comment by esc** [comment]
I have modified the patch to include `--owner` where appropriate.

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch and fixes.

**Pull request merged**